homeassistant/components/utility_meter/select.py

```diff
@@ -0,0 +1,196 @@
+    """Representation of a Tariff selector."""
+
+    def __init__(self, name, tariffs, add_legacy_entities):
+        """Initialize a tariff selector."""
+        self._attr_name = name
+        self._current_tariff = None
+        self._tariffs = tariffs
+        self._attr_icon = TARIFF_ICON
+        self._attr_should_poll = False
+        self._add_legacy_entities = add_legacy_entities
+
+    @property
+    def options(self):
+        """Return the available tariffs."""
+        return self._tariffs
+
+    @property
+    def current_option(self):
+        """Return current tariff."""
+        return self._current_tariff
+
+    async def async_added_to_hass(self):
+        """Run when entity about to be added."""
+        await super().async_added_to_hass()
+
+        await self._add_legacy_entities([LegacyTariffSelect(self.entity_id)])
+
+        state = await self.async_get_last_state()
+        if not state or state.state not in self._tariffs:
+            self._current_tariff = self._tariffs[0]
+        else:
+            self._current_tariff = state.state
+
+    async def async_select_option(self, option: str) -> None:
+        """Select new tariff (option)."""
+        self._current_tariff = option
+        self.async_write_ha_state()
+
+
+class LegacyTariffSelect(Entity):
+    """Backwards compatibility for deprecated utility_meter select entity."""
+
+    def __init__(self, tracked_entity_id):
+        """Initialize the entity."""
+        self._attr_icon = TARIFF_ICON
+        # Set name to influence enity_id
+        self._attr_name = split_entity_id(tracked_entity_id)[1]
+        self.tracked_entity_id = tracked_entity_id
+
+    @callback
+    def async_state_changed_listener(self, event: Event | None = None) -> None:
+        """Handle child updates."""
+        if (
+            state := self.hass.states.get(self.tracked_entity_id)
+        ) is None or state.state == STATE_UNAVAILABLE:
+            self._attr_available = False
+            return
+
+        self._attr_available = True
+
+        self._attr_name = state.attributes.get(ATTR_FRIENDLY_NAME)
+        self._attr_state = state.state
+        self._attr_extra_state_attributes = {
+            ATTR_TARIFFS: state.attributes.get(ATTR_OPTIONS)
+        }
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+
+        @callback
+        def _async_state_changed_listener(event: Event | None = None) -> None:
+            """Handle child updates."""
+            self.async_state_changed_listener(event)
+            self.async_write_ha_state()
+
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, [self.tracked_entity_id], _async_state_changed_listener
+            )
+        )
+
+        # Call once on adding
+        _async_state_changed_listener()
+
+    async def async_select_tariff(self, tariff):
+        """Select new option."""
+        await self.hass.services.async_call(
```

Comment:
Should we throw a deprecation warning when used?

Reply:
With commit 110992d there are warnings logged when the service is called.